src/test/java/com/contrastsecurity/maven/plugin/sdkx/scan/ArtifactScannerTest.java

```diff
@@ -0,0 +1,154 @@
+package com.contrastsecurity.maven.plugin.sdkx.scan;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+import com.contrastsecurity.exceptions.UnauthorizedException;
+import com.contrastsecurity.maven.plugin.sdkx.CodeArtifact;
+import com.contrastsecurity.maven.plugin.sdkx.ContrastScanSDK;
+import com.contrastsecurity.maven.plugin.sdkx.NewCodeArtifactRequest;
+import com.contrastsecurity.maven.plugin.sdkx.Scan;
+import com.contrastsecurity.maven.plugin.sdkx.ScanSummary;
+import com.contrastsecurity.maven.plugin.sdkx.StartScanRequest;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.InOrder;
+
+/**
+ * Unit tests for {@link ArtifactScanner}. Mockist style tests mock the {@link ContrastScanSDK} and
+ * verify that the {@code ArtifactScanner} interacts with it as expected.
+ *
+ * <p>The complexity of monitoring a running scan and retrieving results is encapsulated in {@link
+ * ScanOperation} and not this class. Therefore, these tests cover the happy path and some failure
+ * cases for starting the scan. The more interesting test cases are in {@link ScanOperationTest}.
+ */
+final class ArtifactScannerTest {
+
+  private ScheduledExecutorService executor;
+  private ContrastScanSDK contrast;
+  private ArtifactScanner scanner;
+  private Path artifact;
+  private NewCodeArtifactRequest newCodeArtifactRequest;
+
+  @TempDir Path tmp;
```

Comment:
Minor: this can be a parameter of the `retrieves_results_after_scan_completes` test.

Reply:
But it is also used in the `before()` method